### DISCUSSION TITLE
Persistence in user session?

### DISCUSSION THREAD

**Discussion by ghost** [appium/appium#18862] [Q&A]
I am using Appium with WebDriverIO on Android Chrome and have one question:

1. How can I persist a session so that my authentication details are saved when I start it up again?

I can completely understand "new slate" testing, but that's honestly insane. How often does a user come to my website with a clean slate? Almost always they have information attached and I want to be able to appropriately address that.

And yes, I can easily just re-login. That's no issue.

I have tried noReset: true. I have also tried manually updating the cookies.

**Comment by jlipps** [comment]
What do you mean by "changing the URL"? I don't think I understand what you're saying the issue is.

**Comment by ghost** [comment]
I'm sorry I have edited the original comment. I resolved the url issue (my bad)

My main issue is being able to persist, or create a mock session before entering my page for testing.

**Comment by jlipps** [comment]
Typically with Chrome you'd do something like is described at https://chromedriver.chromium.org/capabilities#h.p_ID_64 to use a custom user data dir. Not sure that works on Chrome on Android.

An industry best practice is to have server side test-only routes that immediately log you in with whatever test state is required (certain user with certain objects created, etc...). So you would call `driver.get('http://yoursite/test/user/foo')` and with no UI automation would be dropped into the test user `foo`'s dashboard or whatever, with all the state set up. This has the advantage of keeping test fixtures/state in the backend, or in your test code, rather than highly volatile and unsafe chrome profiles.

**Comment by ghost** [comment]
Thanks for the prompt & insightful responses.

I believe android chrome user profiles are tied with the Google account and can only be one.

I'm assuming this means that there's no way to use any testing profiles like on PC? Where can I find the code that initializes the chrome session so I can understand how it's able to do so without using my current profile?

**Reply by jlipps** [reply]
That's all part of https://github.com/appium/appium-chromedriver
